src/ascii/port.rs

```diff
@@ -194,6 +181,66 @@ impl<'a, B: Backend> Port<'a, B> {
         }
     }
 
+    /// Convert this port into one that implements `Send` can therefore be sent
+    /// to another thread.
+    ///
+    /// Returns this port as an error if `Send` bounds could not be added. In
+    /// particular if any callbacks are currently set. As such, it is recommended
+    /// to call [`Port::try_into_send`] before setting callbacks if you need to do
+    /// both.
+    ///
+    /// The [`Port::set_packet_handler`] and [`Port::set_unexpected_alert_handler`]
+    /// methods on the new port will have an additional `Send` bound on any function
+    /// used as a callback.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// # use zproto::{ascii::{AnyResponse, Port}, backend::Backend};
+    /// # use std::{any::Any, error::Error, fmt::Debug};
+    /// # fn wrapper<B: Backend + Debug + Send + 'static>(
+    /// #     port: Port<'static, B>
+    /// # ) -> Result<(), Box<dyn Error>> {
+    /// use std::sync::{Arc, Mutex};
+    ///
+    /// let port = Port::open_serial("...")?
+    ///     .try_into_send()  // Required in order to send the port to another thread.
+    ///     .unwrap();
+    /// let port = Arc::new(Mutex::new(port));
+    ///
+    /// let mut handles = vec![];
+    /// for i in 0..10 {
+    ///     let port = port.clone();
+    ///     let handle = std::thread::spawn(move || {
+    ///         let mut guard = port.lock().unwrap();
+    ///         // do something with the port ...
+    ///     });
+    ///     handles.push(handle);
+    /// }
+    ///
+    /// for handle in handles {
+    ///     let _ = handle.join();
+    /// }
+    /// # Ok(())
+    /// # }
+    /// ```
+    pub fn try_into_send(mut self) -> Result<Port<'a, B, SendCallbacks<'a>>, Self> {
```

Comment:
Consider adding an alias for `Port<'a, B, SendCallbacks<'a>>`. Maybe?

Reply:
It could be an associated type. That would reduce clutter in the module level docs.

Reply:
Associated types are only stable for traits. It would be nice to have `Port<'a, B>::Send` but instead I made the type alias `PortSend<'a, B>`. 1e02be4 and b2660db